app/models/gift_card_detail.rb

```diff
@@ -0,0 +1,5 @@
+class GiftCardDetail < ApplicationRecord
+  belongs_to :item
+  validates_uniqueness_of :gift_card_id
+  has_many :gift_card_amount
```

Comment:
a gift card has multiple amounts?  is this to cover spending on the amt? 

just purely a question

Reply:
This is to track the history of the spending of a gift card, otherwise it will be difficult to reconcile reporting, undoing an accidental charge, etc.